lib/gooddata/models/domain.rb

```diff
@@ -432,31 +439,10 @@ def synchronize_clients
     # Runs async process that walks through segments and provisions projects if necessary.
     #
     # @return [Enumerator] Returns Enumerator of results
-    def provision_client_projects(segments = nil)
-      body = if segments
-               {
-                 provisionClientProjects: {
-                   segments: segments.is_a?(Array) ? segments : [segments]
-                 }
-               }
-             end
-
-      res = client.post(segments_uri + '/provisionClientProjects', body)
-      res = client.poll_on_code(res['asyncTask']['links']['poll'])
-      failed_count = GoodData::Helpers.get_path(res, %w(clientProjectProvisioningResult failed count), 0)
-      created_count = GoodData::Helpers.get_path(res, %w(clientProjectProvisioningResult created count), 0)
-      return Enumerator.new([]) if failed_count + created_count == 0 # rubocop:disable Style/NumericPredicate
-      Enumerator.new do |y|
-        uri = GoodData::Helpers.get_path(res, %w(clientProjectProvisioningResult links details))
-        loop do
-          result = client.get(uri)
-          (GoodData::Helpers.get_path(result, %w(clientProjectProvisioningResultDetails items)) || []).each do |item|
-            y << ProvisioningResult.new(item['id'], item['status'], item['project'], item['error'])
-          end
-          uri = GoodData::Helpers.get_path(res, %w(clientProjectProvisioningResultDetails paging next))
-          break if uri.nil?
-        end
-      end
+    def provision_client_projects(options = {})
+      segments = options[:segments]
+
+      data_products(:all).each { |d| d.provision_client_products(segments) }
```

Comment:
Do you mean `provision_client_proJECTS`?

Reply:
This will make `n` API calls where `n == dataproducts.length`. Moreover, the calls will be made sequentially and it will take a longer than optimal time. Can you please reuse the original `provision_client_projects` in both `Domain` and `DataProduct` and add a parameter that will determine whether to make the call to `/domains/:domain/provisionClientProjects` or `/domains/:domain/dataproducts/:dataproduct/provisionClientProjects`?